Code/GWFrames/Extrapolation.py

```diff
@@ -191,23 +224,42 @@ def ReadFiniteRadiusData(ChMass=0.0, filename='rh_FiniteRadii_CodeUnits.h5', Coo
         NWaveforms = len(WaveformNames)
         # Check input data
         if(not ValidateGroupOfWaveforms(f, filename, WaveformNames, LModes)) :
-            raise ValueError("Bad input waveforms in {0}.".format(filename))
+            if EnforceQualityAssurance:
+                raise ValueError("Bad input waveforms in {0}.".format(filename))
         # print("{0} passed the data-integrity tests.".format(filename))
-        stdout.write("{0} passed the data-integrity tests.\n".format(filename)); stdout.flush()
-        Ws = [GWFrames.Waveform() for i in range(NWaveforms)]
-        Radii = [None]*NWaveforms
-        InitialAdmEnergy = f[WaveformNames[0]+'/InitialAdmEnergy.dat'][0,1]
+        if EnforceQualityAssurance:
+            stdout.write("{0} passed the data-integrity tests.\n".format(filename)); stdout.flush()
+        else:
+            stdout.write("Not enforcing data-integrity tests. Continuing extrapolation.\n".format(filename)); stdout.flush()
         DataType = basename(filename).partition('_')[0]
         if('hdot' in DataType.lower()) :
             DataType = GWFrames.hdot
+            SpinWeight = -2;
         elif('h' in DataType.lower()) :
             DataType = GWFrames.h
+            SpinWeight = -2;
         elif('psi4' in DataType.lower()) :
             DataType = GWFrames.Psi4
+            SpinWeight = -2;
+        elif('psi3' in DataType.lower()) :
+            DataType = GWFrames.Psi3
+            SpinWeight = -1;
+        elif('psi2' in DataType.lower()) :
+            DataType = GWFrames.Psi2
+            SpinWeight = 0;
+        elif('psi1' in DataType.lower()) :
+            DataType = GWFrames.Psi1
+            SpinWeight = 1;
+        elif('psi0' in DataType.lower()) :
+            DataType = GWFrames.Psi0
+            SpinWeight = 2;
         else :
             DataType = GWFrames.UnknownDataType
-            raise ValueError("The file '{0}' does not contain a recognizable description of the data type ('h', 'hdot', or 'Psi4').".format(filename))
+            raise ValueError("The file '{0}' does not contain a recognizable description of the data type ('h', 'hdot', 'Psi4', 'Psi3', 'Psi2', 'Psi1', or 'Psi0').".format(filename))
         PrintedLine = ''
+        Ws = [GWFrames.Waveform().SetSpinWeight(SpinWeight) for i in range(NWaveforms)]
```

Comment:
Using `SetSpinWeight` here is ugly and inconsistent.  It would be better to add it as an argument to `ReadFiniteRadiusWaveform`, alongside the `DataType` argument.